From ed26b41ce528f6e4f0800a6b9e3e54b1d5bbf9d6 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Thu, 30 Jan 2025 14:21:35 +0100
Subject: [PATCH] arm64: dts: st: move rng2 node to satisfy base address
 ordering

Move rng2 node before hash1 to satisfy the base address sorting.

Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Change-Id: I4d076506a66ac6f9d61856e8e929d0dd8fa1cacf
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/433140
Domain-Review: Eric FOURMONT <eric.fourmont-ext@st.com>
Tested-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Reviewed-by: Eric FOURMONT <eric.fourmont-ext@st.com>
---
 arch/arm64/boot/dts/st/stm32mp211.dtsi | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/arch/arm64/boot/dts/st/stm32mp211.dtsi b/arch/arm64/boot/dts/st/stm32mp211.dtsi
index fccf708d27d4..1a728062e33f 100644
--- a/arch/arm64/boot/dts/st/stm32mp211.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp211.dtsi
@@ -1590,6 +1590,17 @@ channel@17 {
 				};
 			};
 
+			rng2: rng@42020000 {
+				compatible = "st,stm32mp21-rng";
+				reg = <0x42020000 0x0 0x400>;
+				clocks = <&clk_rcbsec>, <&rcc CK_BUS_RNG2>;
+				clock-names = "rng_clk", "rng_hclk";
+				resets = <&rcc RNG2_R>;
+				access-controllers = <&rifsc 93>;
+				power-domains = <&d1_pd>;
+				status = "disabled";
+			};
+
 			hash1: hash@42030400 {
 				compatible = "st,stm32mp13-hash";
 				reg = <0x42030400 0x0 0x400>;
@@ -1603,17 +1614,6 @@ hash1: hash@42030400 {
 				status = "disabled";
 			};
 
-			rng2: rng@42020000 {
-				compatible = "st,stm32mp21-rng";
-				reg = <0x42020000 0x0 0x400>;
-				clocks = <&clk_rcbsec>, <&rcc CK_BUS_RNG2>;
-				clock-names = "rng_clk", "rng_hclk";
-				resets = <&rcc RNG2_R>;
-				access-controllers = <&rifsc 93>;
-				power-domains = <&d1_pd>;
-				status = "disabled";
-			};
-
 			iwdg1: watchdog@44010000 {
 				compatible = "st,stm32mp1-iwdg";
 				reg = <0x44010000 0x0 0x400>;
-- 
2.50.1

